test/tbb/test_partitioner.cpp

```diff
@@ -77,8 +101,7 @@ template <typename PerBodyFunc> float test(PerBodyFunc&& body) {
                 tbb::static_partitioner()
             );
         });
-        // TODO:
-        //   - Consider introducing an observer to guarantee the threads left the arena.
+        observer.wait_leave();
```

Comment:
I would add the comment so that future readers better understand why the observer was introduced. Something like "_To avoid tasks stealing at a beginning of the parallel algorithm, the test waits when the threads leave the arena so that on the next iteration they have tasks assigned in their mailboxes and, thus, don't need to search for a work to do in other task pools_"

Reply:
Added.